moveit_ros/visualization/handeye_calibration_rviz_plugin/include/moveit/handeye_calibration_rviz_plugin/handeye_control_widget.h

```diff
@@ -0,0 +1,242 @@
+    tf_buffer_.reset();
+    solver_.reset();
+    solver_plugins_loader_.reset();
+    move_group_.reset();
+    planning_scene_monitor_.reset();
+  }
+
+  void loadWidget(const rviz::Config& config);
+  void saveWidget(rviz::Config& config);
+
+  void setTFTool(rviz_visual_tools::TFVisualToolsPtr& tf_pub);
+
+  void addPoseSampleToTreeView(const geometry_msgs::TransformStamped& cTo, const geometry_msgs::TransformStamped& bTe,
+                               int id);
+
+  bool loadSolverPlugin(std::vector<std::string>& plugins);
+
+  bool createSolverInstance(const std::string& plugin_name);
+
+  void fillSolverTypes(const std::vector<std::string>& plugins);
+
+  std::string parseSolverName(const std::string& solver_name, char delimiter);
+
+  bool takeTranformSamples();
+
+  bool solveCameraRobotPose();
+
+  bool frameNamesEmpty();
+
+  bool checkJointStates();
+
+  void computePlan();
+
+  void computeExecution();
+
+Q_SIGNALS:
+
+  void sensorPoseUpdate(double x, double y, double z, double rx, double ry, double rz);
+
+public Q_SLOTS:
+
+  void UpdateSensorMountType(int index);
+
+  void updateFrameNames(std::map<std::string, std::string> names);
+
+private Q_SLOTS:
+
+  void takeSampleBtnClicked(bool clicked);
+
+  void resetSampleBtnClicked(bool clicked);
+
+  void saveCameraPoseBtnClicked(bool clicked);
+
+  void planningGroupNameChanged(const QString& text);
+
+  void saveJointStateBtnClicked(bool clicked);
+
+  void loadJointStateBtnClicked(bool clicked);
+
+  void autoPlanBtnClicked(bool clicked);
+
+  void autoExecuteBtnClicked(bool clicked);
+
+  void autoSkipBtnClicked(bool clicked);
+
+  void planFinished();
+
+  void executeFinished();
+
+private:
+  // **************************************************************
+  // Qt components
+  // **************************************************************
+
+  QTreeView* sample_tree_view_;
+  QStandardItemModel* tree_view_model_;
+
+  QComboBox* calibration_solver_;
+
+  // Load & save pose samples and joint goals
+  QPushButton* save_joint_state_btn_;
+  QPushButton* load_joint_state_btn_;
+  QPushButton* save_camera_pose_btn_;
+
+  // Manual calibration
+  QPushButton* take_sample_btn_;
+  QPushButton* reset_sample_btn_;
+
+  // Auto calibration
+  QComboBox* group_name_;
+  QPushButton* auto_plan_btn_;
+  QPushButton* auto_execute_btn_;
+  QPushButton* auto_skip_btn_;
+
+  // Progress of finished joint states for auto calibration
+  ProgressBarWidget* auto_progress_;
+
+  QFutureWatcher<void>* plan_watcher_;
+  QFutureWatcher<void>* execution_watcher_;
+
+  // **************************************************************
+  // Variables
+  // **************************************************************
+
+  mhc::SENSOR_MOUNT_TYPE sensor_mount_type_;
+  std::map<std::string, std::string> frame_names_;
+  // Transform samples
+  std::vector<Eigen::Isometry3d> effector_wrt_world_;
+  std::vector<Eigen::Isometry3d> object_wrt_sensor_;
+  std::string from_frame_tag_;
```

Comment:
This variable should be initialized, otherwise the "Save Camera Pose" button gives an error if the user did not actively click the dropdown box.

I don't mind if it is done here as in the suggestion of in the cpp along with the rest [here](https://github.com/ros-planning/moveit/pull/1560/files#diff-9c50f963bf025314f7ff7d834be1f3f2R95), but this took some error to find and was a confusing error.
```suggestion
  std::string from_frame_tag_ = "base";
```